whisper/whisperv6/whisper.go

```diff
@@ -983,9 +981,16 @@ func validatePrivateKey(k *ecdsa.PrivateKey) bool {
 	return ValidatePublicKey(&k.PublicKey)
 }
 
-// validateSymmetricKey returns false if the key contains all zeros
-func validateSymmetricKey(k []byte) bool {
-	return len(k) > 0 && !containsOnlyZeros(k)
+// validateSymmetricKey returns false if the key contains all zeros,
+// which is a simplest and the most common bug.
+func validateRandomData(k []byte, expectedSize int) bool {
+	if len(k) != expectedSize {
+		return false
+	}
+	if expectedSize > 4 && containsOnlyZeros(k) {
```

Comment:
because i have already seen a test failing because it tried to generate a padding of 1 byte, and came up with 0 (which is completely legitimate random value). So, in case of small padding it is fine. The keys and nonces are much larger anyway.

Reply:
But you are right, 4 is too much. Changed to 3. This way we will probably have one failed test in our entire lifetime.